arduino/src/main.cpp

```diff
@@ -20,40 +24,73 @@ void setup(){
     while(WiFi.status() != WL_CONNECTED){
         Serial.print(".");
         delay(100);
+        
     }
-
-    
+    pinMode(LED,OUTPUT);
+    pinMode(BUTTON,INPUT_PULLUP);
 }
-
-void loop() {
-    // Odczytaj stan GPIO
-   int gpioState = 10 ;
+int send_status_request(int buttonState) 
+{
+    
     DynamicJsonDocument jsonDoc(200);
-    jsonDoc["gpioState"] = gpioState;
+    jsonDoc["place"] =  MIEJSCE;
+    if(buttonState==LOW)
+    {
+        jsonDoc["state"] = "hanged";
+    }
+    else
+    {
+        jsonDoc["state"]="empty";
+    }
+    jsonDoc["board_id"] = BOARD_ID;
     String payload;
-    serializeJson(jsonDoc,payload);
+    serializeJson(jsonDoc, payload);
     // Wyślij żądanie POST na serwer Flask
     HTTPClient http;
-    http.begin(serverAddress);
+    http.begin(SERVER_ADDRESS);
     http.addHeader("Content-Type", "application/json");
-     int httpResponseCode = http.sendRequest("GET", payload);
-    if(httpResponseCode > 0)
-    {
-        Serial.printf("HTTP Response code: %d\n",httpResponseCode );
-        String response = http.getString();
-        DynamicJsonDocument jsonDoc(200);
-        deserializeJson(jsonDoc, response);
-        String success = jsonDoc["success"].as<String>();
-        Serial.println(response);
+    int httpResponseCode = http.sendRequest("POST", payload);
+    http.end();
+    delay(1000);
+    return httpResponseCode;
+}
 
+
+void loop() 
+{   
+    int buttonState = digitalRead(BUTTON);
+    if(buttonState==LOW)   
+    {
+        digitalWrite(LED,HIGH);
     }
     else
     {
-        Serial.printf("HTTP Request failed %s\n", http.errorToString(httpResponseCode).c_str());
-
+        digitalWrite(LED,LOW);
+    }
+    int httpResponseCode = send_status_request(buttonState);
+    if(httpResponseCode < 0)
+    {
+        int buttonState = digitalRead(BUTTON);
+        if(buttonState==LOW) 
+        { 
+            while (1) 
+            {
+                digitalWrite(LED, HIGH);
+                delay(500);
+                digitalWrite(LED, LOW);
+                delay(500);
+                buttonState = digitalRead(BUTTON);
+                httpResponseCode = send_status_request(buttonState);
+                if(httpResponseCode>0 )
+                {
+                    break;
+                }

```

Comment:
Ten sam problem co w poprzednim komentarzu. Kiedy ten kod http ma wartość równą, mniejszą zeru?

Reply:
kiedy nie łączy sie z serwerem

Reply:
wtedy wartośc jest -1

Reply:
Ale jeśli połączy się z serwerem, ale odwoła się do niewłaściwego URLa to dostanie kod 404. Takie przypadki też trzeba ogarnąć.

Reply:
i wtedy dioda w ogóle nie powinna się świecić? tak żeby odróżnić to od stanu w którym serwer się zawiesił ?

Reply:
Dla uproszczenia wtedy też zakładamy że jest awaria, niech haczyk zachowuje się tak samo, jak w wypadku kiedy wcale nie nawiąże kontaktu. 
Tylko część kodów HTTP oznacza sukces.

Reply:
> Tylko część kodów HTTP oznacza sukces

Weź zajrzyj do linków o kodach błędów, które ci podesłałem przeczytaj je ze zrozumieniem i wtedy popraw kod, aby obsługiwał właściwe wartości. Niby dodałeś kod 404, ale czy sądzisz, że to jest jedyny istniejący kod błędu?

Reply:
dobra przejrzałem i zmieniłem kod.

Reply:
teraz chyba wszystko jest już dobrze.  
Potem jeszcze raz spojrzę na ten kod i ci go przyklepie. Powiedz  mi tylko, czy testowałeś go już na urządzeniu?

Reply:
tak

Reply:
swoją droga chyba będę musiał zmienić kod od serwera i readme bo tam zmienia wartość na empty po 10 sekundach bez otrzymania informacji od płytki.

Reply:
Tak, będziesz musiał, jak chcesz to już możesz stworzyć na to nowego pull requesta